subiquity/server/server.py

```diff
@@ -466,6 +476,79 @@ async def apply_autoinstall_config(self, context):
             await controller.apply_autoinstall_config()
             await controller.configured()
 
+    def _strip_controller_keys(self, autoinstall_config: dict) -> dict:
+        return_dict: dict = copy.deepcopy(autoinstall_config)
+
+        for controller in self.controllers.instances:
+            return_dict.pop(controller.autoinstall_key, None)
+
+        return return_dict
+
+    def _enforce_top_level_keys(self, base_config: dict) -> dict:
+        # Assumes base_config is the result of stripping away all
+        # controller related sections (via _strip_controller_keys)
+
+        top_level_keys: set[str] = set(base_config.keys())
+        valid_keys: set[str] = set(self.base_schema["properties"].keys())
+        bad_keys: set[str] = top_level_keys - valid_keys
+
+        # Return early if no bad keys
+        if not bad_keys:
+            return base_config
+
+        # If version early enough, only warn
+        version: int = base_config.get("version", None)
+
+        if not version:
+            raise AutoinstallValidationError(
+                section="top-level keys",
+                message="top-level key 'version' is missing",
+            )
+
+        if version <= 1:
+            for key in bad_keys:
+                log.warning(f"Unrecognized top-level autoinstall key {key!r}")
+
+            log.warning(
+                "Unrecognized keys may cause autoinstall to crash in future versions"
+            )
```

Comment:
Currently all this will do is print with `WARNING` to the log file. Do we see a need to use something like `warning.warn` to print to stderr instead? (Or stdout where the main server process is run)

Reply:
> crash

I don't love this word, as the word `crash` will be seen to mean a defect. 

> stderr

No, that's not quite right.  We need two cases to show the fault
* interactive - we need to show the results in a dialog that indicates that the autoinstall is bad.  The existing crash dialog is a good clue, but we won't reuse it directly because this is a non-defect case.
* non-interactive - output needs to go to the same place as the rest of the UI, which may be on a serial console etc.  Follow the existing context logging for that.

Also warn is the wrong mindset in that if we're supposed to be autoinstalling and we have stopped, then that is fatal to the program.

> may cause

We have a distinct plan, it will cause, so let's set that text accordingly.